src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,393 @@
+import { AbstractDataExportStrategy } from './AbstractDataExportStrategy';
+
+export abstract class AbstractComponentDataExportStrategy extends AbstractDataExportStrategy {
+  canViewStudentNames: boolean;
+  includeOnlySubmits: boolean;
+  includeStudentNames: boolean;
+  workSelectionType: string;
+
+  constructor(protected nodeId: string, protected component: any, additionalParams: any) {
+    super();
+    this.canViewStudentNames = additionalParams.canViewStudentNames;
+    this.includeOnlySubmits = additionalParams.includeOnlySubmits;
+    this.includeStudentNames = additionalParams.includeStudentNames;
+    this.workSelectionType = additionalParams.workSelectionType;
+  }
+
+  abstract generateComponentWorkRow(
+    component: any,
+    columnNames: string[],
+    columnNameToNumber: any,
```

Comment:
Define new ColumnNameToNumber interface?
```suggestion
    columnNameToNumber: ColumnNameToNumber,
```

Reply:
What would the ColumnNameToNumber interface look like? columnNameToNumber is an object where the key is a column name string and the value is a number.